File: articles/reading-notes/head-first-design-pattern/code/02-obsever-pattern/03-KVO/03-KVO/ViewController.m
//
//  ViewController.m
//  03-KVO
//
//  Created by zhouke on 2018/6/21.
//  Copyright © 2018年 zhouke. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (nonatomic, assign) NSInteger count;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    [self addObserver:self forKeyPath:@"count" options:NSKeyValueObservingOptionOld | NSKeyValueObservingOptionNew context:nil];
}

- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary<NSKeyValueChangeKey,id> *)change context:(void *)context
{
    NSLog(@"%@", change);
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    self.count = 10;
}

@end

File: docs/articles/reading-notes/Graphic-Design-Patterns/code/08-Strategy/08-Strategy/ViewController.swift
//
//  ViewController.swift
//  08-Strategy
//
//  Created by zhouke on 2018/11/2.
//  Copyright © 2018 zhouke. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        let context = Context()
        
        let concreteStrategyA = ConcreteStrategyA()
        context.strategy = concreteStrategyA
        context.algorithm()
        
        let concreteStrategyB = ConcreteStrategyB()
        context.strategy = concreteStrategyB
        context.algorithm()
        
    }


}

